x-pack/solutions/observability/plugins/observability/public/components/rule_condition_chart/helpers.ts

```diff
@@ -18,7 +18,7 @@ export const getLensOperationFromRuleMetric = (metric: GenericMetric): LensOpera
   const { aggType, field, filter = '' } = metric;
   let operation: string = aggType;
   const operationArgs: string[] = [];
-  const escapedFilter = filter.replace(/'/g, "\\'");
+  const escapedFilter = filter.replace(/\\/g, '\\\\').replace(/'/g, "\\'");
```

Comment:
Can we add a test for this helper that fails before this fix?

Reply:
@maryam-saeidi I have already added a test for this helper in another PR. This PR is security-related only.

Reply:
Oh, really? Thanks for sharing. How does it work if we add a test before having the actual fix? I am not familiar with how the security scanning works.